### PR TITLE
Add `Assert.type_relation` for `<:` and `!<:` expressions in `assert:` macro

### PR DESCRIPTION
**Pull request by tonchis**
Issue: #88 

This PR addresses creates a special case `Assert.type_relation` function for the `assert: foo <: Some` (and `!<:`) operators because the right-hand side is a type expression, not a value expression, so we can't treat it the same as other operators in the language.

Error messages for this macro look like this:
image

### CODE REVIEW

File: src/savi/compiler/macros.cr [hunk 1]
```diff
@@ -638,6 +646,54 @@ class Savi::Compiler::Macros < Savi::AST::CopyOnMutateVisitor
...
   end
 
+  def visit_assert_type_relation(node : AST::Relate, expr : AST::Relate)
+    orig = node.lhs.as(AST::Identifier)
+    lhs = expr.lhs
+    op = expr.op
+    rhs = expr.rhs
+    
+    # Use a hygienic local to explicitly hold the expressions as box.
+    # This also helps to refer to them multiple times without evaluating them again.
```
Comment (jemc) [inline src/savi/compiler/macros.cr:656]: In this case we are not using an explicit `box` type.

```suggestion
    # Use a hygienic local to refer to it multiple times without evaluating again.
```

**Review by jemc — APPROVED** [review]
One small fix to a copied comment:

**Comment by jemc** [comment]
Thanks!

**Comment by tonchis** [comment]
🙌

**Pull request merged**